### PR TITLE
Upgrade to steal and steal-tools 0.16.6

### PR DESCRIPTION
**Pull request by pmgmendes**
Related to issue https://github.com/bitovi/documentjs/issues/270

### CODE REVIEW

**Comment by pmgmendes** [comment]
Sorry the multiple commits. The intended change is the latest - https://github.com/bitovi/documentjs/pull/271/commits/17b8974221330e9d5fba9a4b182224473637ff55

File: package.json [hunk 1]
```diff
@@ -45,8 +45,8 @@
     "md5": "^2.0.0",
     "minimatch": "~1.0.0",
     "q": "~1.0.1",
-    "steal": "0.13.X",
-    "steal-tools": "0.13.X",
+    "steal": "0.16.6",
+    "steal-tools": "0.16.6",
```
Comment (chasenlehara) [inline package.json]: In case we ever do another 0.16 release, I would make these `^0.16.6`

Comment (pmgmendes) [inline package.json]: After adding engineDependencies for Node 6 and 7 the [build still fails](https://travis-ci.org/pmgmendes/documentjs/jobs/225398042) due 6 failing tests. The troubleshooting points me to steal.js, getScriptOptions(), line 6980
```
...
var optionName =
	camelize( attr.nodeName.indexOf("data-") === 0 ?
		attr.nodeName.replace("data-","") :
		attr.nodeName );
...
```
not sure though..
Adding the caret range specifier causes 1 test to fail in the [node 0.10 build](https://travis-ci.org/pmgmendes/documentjs/jobs/225398040). Still didn't got into it.

**Comment by justinbmeyer** [comment]
Maybe @leoj3n can look into this on wednesday?

**Comment by leoj3n** [comment]
The failing tests look very similar (identical) to the steal(-tools) + zombie issues encountered in:

https://github.com/stealjs/steal/issues/1177

@matthewp and I hacked on that issue today, culminating in a stopgap change in steal to get past the bug, but it's still not loading (something else must be happening in the bundle—some code is not working, and the error is not bubbling). Matthew may look more in the morning tomorrow if he has time, but he mentioned that he doesn't want to officially support zombie (at least not right now).

I will definitely apply anything we learn to this issue, especially if we find a solution.

**Comment by justinbmeyer** [comment]
What was the stop-gap?  Do we know what was going wrong?  cc @matthewp

**Comment by matthewp** [comment]
No, we don't know yet. I'm looking into it tomorrow.

**Comment by leoj3n** [comment]
Reduced test case: https://github.com/leoj3n/generate-temp#generate-temp

**Comment by m-mujica** [comment]
Master is already using 0.16.x

**Pull request closed without merging**